### ISSUE TITLE
Integrate appservices sync manager component

### ISSUE THREAD

**Issue by lougeniaC64** [mozilla-mobile/firefox-ios#13278]
This ticket represents the work required by the A-S team to integrate the rust sync manager component into iOS. In order to make reviewing the code easier, this work will be split out into a number of PRs which include the following work:

- Adding qualified names to the current `SyncReason`, `SyncResult`, and `SyncManager` types so they do not conflict with the types introduced by the rust component
  - Note: The native types listed above will ultimately be replaced by the  types introduced by the component once the Nimbus rollout is complete
- Adding supporting functions for sync manager integration including A-S API functions and logic to report the telemetry returned by the sync manager component's `sync`function
- Adding the `RustSyncManager` class
  - This class will eventually replace `BrowserSyncManager`
- Adding the Nimbus rollout logic and updating `BrowserProfile` to use `RustSyncManager` when the experiment is enabled 

Once the Sync Manager Nimbus experiment is complete, a separate PR will be created to remove the old sync code.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-5790)
